### PR TITLE
plan: fix a panic bug when involved join and order by.

### PR DESCRIPTION
**Pull request by hanfei1991**
Consider the query:
`select * from a join (select * from t limit 1) b on a.c = b.c order by b.c`
the final plan will be Join{table(a), table(b)->Projection->Limit->Order by b.c} -> Projection.
Firstly, we calucate the b.c's index as 1. When we push "order by" past join, we should update b.c's index to avoid panic.
@shenli @coocood @zimulala @XuHuaiyu PTAL

### CODE REVIEW

File: plan/physical_plan_builder.go [hunk 1]
```diff
@@ -549,6 +567,8 @@ func (p *Join) convert2PhysicalPlanRight(prop *requiredProperty, innerJoin bool)
 	rProp := prop
 	if !allRight {
 		rProp = &requiredProperty{}
+	} else {
```
Comment (XuHuaiyu) [inline plan/physical_plan_builder.go]: why not handle this in `convert2PhysicalPlanLeft`?

Comment (hanfei1991) [inline plan/physical_plan_builder.go]: the columns in left plan keeps same index with the columns in join.

File: plan/physical_plan_builder.go [hunk 2]
```diff
@@ -517,6 +518,23 @@ func (p *Join) convert2PhysicalPlanLeft(prop *requiredProperty, innerJoin bool)
 	return resultInfo, nil
 }
 
+// replaceColsInPropBySchema replaces the columns in originSchema with the new columns in newSchema.
+func replaceColsInPropBySchema(prop *requiredProperty, originSchema expression.Schema, newSchema expression.Schema) *requiredProperty {
```
Comment (coocood) [inline plan/physical_plan_builder.go]: Only use one argument is enough for `convert2PysicalPlanRight`.

File: plan/physical_plan_builder.go [hunk 3]
```diff
@@ -517,6 +518,23 @@ func (p *Join) convert2PhysicalPlanLeft(prop *requiredProperty, innerJoin bool)
...
+	newProps := make([]*columnProp, 0, len(prop.props))
+	for _, p := range prop.props {
+		idx := originSchema.GetIndex(p.col)
+		if idx == -1 {
+			log.Errorf("Can't find column %s in schema", p.col)
```
Comment (hanfei1991) [inline plan/physical_plan_builder.go]: This log will record the error info info when panic happens.

**Comment by XuHuaiyu** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**